packages/web3/src/contract/contract.ts

```diff
@@ -1725,14 +1725,15 @@ function getBytecodeTemplate(
   // For the default TxScript main function
   const numberOfMethods = '01'
   const isPublic = '01'
-  const modifier = usePreapprovedAssets ? '03' : '00'
+  const scriptUseApprovedAssets = attoAlphAmount !== undefined || tokens !== undefined
```

Comment:
Question: In what situations does a contract method use preapproved assets, but the script does not use preapproved assets?

Reply:
That's impossible, but it's possible that the method does not use preapproved assets, but the script use preapproved assets. E.g. the withdraw function of token faucet.

Reply:
Got it, thanks for the explanation